File: New folder/python/binarySearch.py
def binSearch(values,target):
    low=0
    high=len(values)-1
    while low<=high:
        print("low is ->",low)
        mid=int((low+high)/2)
        print("mid is ->",mid)
        print("high is ->",high)
        if values[mid]==target:
            return mid
        elif values[mid]>target:
            high=mid-1
        else:
            low=mid+1
    return -1

if __name__=="__main__":
    values=list(map(int,input("Enter the list > ").split()))
    target=int(input("Enter the target value > "))
    idx=binSearch(sorted(values),target)
    if idx!=-1:
        print("Target was found at",idx)
    else:
        print("Target was not found")

File: cpp/binarySearch.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int binSearch(int *arr,int n,int target){
    int l=0,r=n-1;
    while(l<=r){
        int mid=(l+r)/2;
        cout << " l is -> " << l;
        cout << " mid is -> " << mid;
        cout << " r is -> " << r;
        cout << endl;
        if(arr[mid]==target){
            return mid;
        }
        else if(arr[mid]>target){
            r=mid-1;
        }
        else{
            l=mid+1;
        }
    }
    return -1;
}

int main(){
    int n;
    cout << "Enter the size of the array -> ";
    cin >> n;

    int * arr=new int[n];
    cout << "Enter the elements -> ";
    for(int i=0;i<n;i++){
        cin >> arr[i];
    }

    int target;
    cout << "Enter the target value -> ";
    cin >> target;
    int idx=binSearch(arr,n,target);
    if(idx==-1) cout << "Element not found" << endl;
    else cout << "Element found at " << idx << endl;
}